Automattic/jetpack
# cli: Fix user filename input with chars like `#`

#### Changes proposed in this Pull Request:

In CommonJS, it's straightforward to combine user input into a relative
file name with `path.join( __dirname, input )`. But in ESM `__dirname`
is unavailable. Instead we have `import.meta.url`, which is a
file-protocol URL.

The simple fix, `new URL( input, import.meta.url )`, works in many cases
but will probably fail if the new file path contains characters like `#`
or `?` that are special in URLs. In these cases it's probably better to
pass `import.meta.url` through `fileURLToPath()` so we can `path.join`
it.

#### Jetpack product discussion


None

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Make sure `jetpack generate` still works right.
* Try `jetpack changelog add`, particularly in cases where the name without the `#` already exists so you want to do a `#2` name.
